demo/tests/test_conditions.py

```diff
@@ -251,6 +251,106 @@ def get_form_class(self, formidable, role):
         return get_dynamic_form_class_from_schema(schema)
 
 
+class DropdownConditionsTestCase(TestCase):
+
+    def get_form_class(self, formidable, role):
+        return get_dynamic_form_class(formidable, role)
+
+    def setUp(self):
+        conditions_schema = [
+            {
+                'name': 'Show a and b if value "ab" selected',
+                'action': 'display_iff',
+                'fields_ids': ['a', 'b', ],
+                'tests': [
+                    {
+                        'field_id': 'main_dropdown',
+                        'operator': 'eq',
+                        'values': ['ab'],
+                    }
+                ]
+            },
+            {
+                'name': 'Show b if value "b" selected',
+                'action': 'display_iff',
+                'fields_ids': ['b'],
+                'tests': [
+                    {
+                        'field_id': 'main_dropdown',
+                        'operator': 'eq',
+                        'values': ['b'],
+                    }
+                ]
+            }
+        ]
+
+        class DropDownForm(FormidableForm):
```

Comment:
oh, now I see... we're still in the "setUp" here... I've always find this kind of pattern a bit confusing. never mind...

Reply:
Could you advise a better solution than making a local class in the `setUp` method?

Reply:
why should this class being local in the first place? it's like a fixture, it's okay to declare it here, but it doesn't **have** to be local, IMO.
I think it confuses *me* when I browse the code without too much care: I see a ``class``, then a ``def``, then a ``class``, so I think that I'm out of the first class.

Reply:
Maybe we should think about the pytest too? https://docs.pytest.org/en/latest/fixture.html
I really this module for the fixtures. And as for me it's cleaner solution than just import/open files